### PR TITLE
chore: [DependencyUpdate] Update XSUAA lib to 3.3.3 (Reloaded)

### PR DESCRIPTION
**Pull request by newtork**
- [x] Update XSUAA lib to 3.3.3
- [x] Release notes

Challenges:
* `java-security-test` brings "Jetty 12" and relies on the API on runtime.
* `wiremock` depends on "Jetty 11" and `wiremock-standalone` shades(!) "Jetty 11".
* `wiremock` + `java-security-test` ("Jetty 12") is not supported
* `wiremock-standalone` + `java-security-test` ("Jetty 12") works, but compatibility is unclear. Accepted risk.

### CODE REVIEW

File: cloudplatform/connectivity-destination-service/pom.xml [hunk 1]
```diff
@@ -228,6 +217,7 @@
 					<ignoredUsedUndeclaredDependencies>
 						<!-- include resilience via the aggregator to ensure the default implementation is present -->
 						<ignoredUsedUndeclaredDependency>com.sap.cloud.sdk.cloudplatform:resilience-api</ignoredUsedUndeclaredDependency>
+						<ignoredUsedUndeclaredDependency>org.wiremock:wiremock-standalone</ignoredUsedUndeclaredDependency>
```
Comment (newtork) [inline cloudplatform/connectivity-destination-service/pom.xml:220]: Relying on transitive `wiremock-standalone` from `java-security-test` is not good.

However declaring it explicitly with a specific version, will risk future incompatibilities with `java-security-test` 🤷‍♂️

Comment (CharlesDuboisSAP) [reply cloudplatform/connectivity-destination-service/pom.xml:220]: Could you add it as a comment above the <ignore>?

File: cloudplatform/connectivity-destination-service/src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (newtork) [inline cloudplatform/connectivity-destination-service/src/main/webapp/WEB-INF/web.xml:1]: (Major)

Please move file to non-productive code.

**Comment by KavithaSiva** [comment]
I would also additionally want the `jetty-bom` in our archetypes to be updated to `12.0.5` .

**Review by KavithaSiva — CHANGES_REQUESTED** [review]
I would also additionally want the `jetty-bom` in our archetype to be updated to `12.0.5`  with a compatibility note that people adding `wiremock` would have to use `wiremock-standalone` for it to work.

File: cloudplatform/connectivity-destination-service/src/main/webapp/WEB-INF/web.xml (hunk 2 above)
Comment (newtork) [reply cloudplatform/connectivity-destination-service/src/main/webapp/WEB-INF/web.xml:1]: image

File does not need to be moved. It's not part of the packaged JAR!

**Comment by newtork** [comment]
> I would also additionally want the `jetty-bom` in our archetype to be updated to `12.0.5` with a compatibility note that people adding `wiremock` would have to use `wiremock-standalone` for it to work.

It's working :+1:
https://sdk-v2.cpe.c.eu-de-1.cloud.sap/job/end-to-end-tests/view/Cloud%20SDK%20v5/job/v5-scp-cf-spring-xsuaa/

**Pull request merged**